junction/src/main/java/org/killbill/billing/junction/plumbing/billing/DefaultBillingEventSet.java

```diff
@@ -78,23 +70,16 @@ public List<UUID> getSubscriptionIdsWithAutoInvoiceOff() {
 
     @Override
     public Map<String, Usage> getUsages() {
-        final Iterable<Usage> allUsages = Iterables.concat(Iterables.transform(this, new Function<BillingEvent, List<Usage>>() {
-            @Override
-            public List<Usage> apply(final BillingEvent input) {
+        final Map<String, Usage> result = new HashMap<>();
+        this.stream()
+            .map(input -> {
                 try {
                     return input.getUsages();
                 } catch (final CatalogApiException e) {
                     throw new IllegalStateException(String.format("Failed to retrieve usage section for billing event %s", input), e);
```

Comment:
Do you have any idea what kind of stack trace we get if this exception is thrown here?

Reply:
I had no idea..  :sweat_smile: . I added unit test to make sure [here](https://github.com/killbill/killbill/pull/1668/files#diff-da2e2dda69a816116e94732c6c006c8325f3be70b4f244a13f47cf0511cad1a1).

Reply:
👍